### PR TITLE
Add Associations

### PR DESCRIPTION
**Pull request by nca6109**
### Associations
Added associations between: 
- User & Panel Set (One-To-Many)
- Panel Set & Panel (One-To-Many)
- Panel Set & Hook (One-To-One)
- Panel & Hook (One-To-Many)

#### Eager loading examples
Associations were tested with eager loading calls. Test runs below

- User & Panel Set
     - User -> Panel Set
        - User-PanelSets_Code
        - User-PanelSets
     - Panel Set -> User
        - PanelSet-User_Code
        - PanelSet-User
   
- Panel Set & Panel
     - Panel Set -> Panel
        - PanelSet-Panel_Code
        - PanelSet-Panel
     - Panel -> Panel Set
        - Panel-PanelSet_Code
        - Panel-PanelSet
     
- Panel Set & Hook
     - Panel Set -> Hook
        - PanelSet-Hook_Code
        - PanelSet-Hook
     - Hook -> Panel Set
        - Hook-PanelSet_Code
        - Hook-PanelSet
        
- Panel & Hook
     - Panel -> Hook
        - Panel-Hook_Code
        - Panel-Hook
     - Hook -> Panel
        - Hook-Panel_Code
        - Hook-Panel

### CODE REVIEW

**Review by BlckHawker — COMMENTED** [review]
Looks good on paper. Can you refactor the controllers that require associations, such as `_createPanelSetController`?

**Review by wct9667 — APPROVED** [review]
Approved!

File: src/services/panelService.ts [hunk 1]
```diff
@@ -93,9 +93,9 @@ const getPanelBasedOnPanelSetAndIndex = (sequelize : Sequelize) => async (index
  * @returns {object[]} An array of objects with id, image, index properties
  */
 const getPanelsFromPanelSetID = (sequelize : Sequelize) => async (panel_set_id: number) => {
-
     // Find all panels on requested panelSet 
-    const panels = await sequelize.models.panel.findAll({ where: { panel_set_id } }) as IPanel[];
+    const panel_set = await sequelize.models.panel_set.findByPk(panel_set_id, {include: sequelize.models.panel}) as IPanelSet;
+    const panels = panel_set.panels as IPanel[];
```
Comment (BlckHawker) [inline src/services/panelService.ts]: This should probably be changed to return an empty array, and return a 404
image

**Review by BlckHawker — CHANGES_REQUESTED** [review]
Nice job. Only found one bug

File: src/services/panelService.ts (hunk 1 above)
Comment (nca6109) [reply src/services/panelService.ts]: Fixed the issue here and the same one in `getAllPanelSetsFromUser`

**Review by BlckHawker — APPROVED** [review]
sonic-boom-knuckles

**Pull request merged**